### PR TITLE
Creation of PDB code to SMILES with protonation

### PR DESCRIPTION
**Pull request by 404random**
## Description
PDB 3 or 4 letter code to canonical isomerism SMILES
## Todos
-improve testing
## Questions
-Generate 3d structure in this file or different file?

## Status
-ready to deploy

### CODE REVIEW

**Comment by 404random** [comment]
@rafwiewiora, @pgrinaway Could you two review the code changes that I made so far? Setup.py works, although the packages openbabel and pypdb need to be added manually to the environment.

**Comment by jchodera** [comment]
You can add their names to the "Reviewers" column in the upper right to formally request reviews!

File: Ligandr/NameToSMILES.py [hunk 1]
```diff
@@ -1,34 +1,55 @@
...
+    
+    pH : double, optional default = 7.0
+        The pH at which protonation should occur
+    
+    Returns
+    --------
+    protonated_smiles : str
+        The SMILES string representing the protonated ligand
+    """
+    mol = openbabel.OBMol()
```
Comment (pgrinaway) [inline Ligandr/NameToSMILES.py]: Do you want to use OpenBabel or RDKit or OpenEye? Any is fine, but be aware that the way that SMILES are canonicalized will differ based on the package.

Comment (jchodera) [reply Ligandr/NameToSMILES.py]: We should use RDKit if possible, OpenEye if necessary.

Let's not use OpenBabel.

Comment (404random) [reply Ligandr/NameToSMILES.py]: I can't seem to find a way to assign different protonation states based on pH to molecules outside of openbabel.

Comment (pgrinaway) [reply Ligandr/NameToSMILES.py]: Are you trying to predict pKas, or just enumerate protonation states?

Comment (pgrinaway) [reply Ligandr/NameToSMILES.py]: Or do you just need smiles with explicit protons?

Comment (404random) [reply Ligandr/NameToSMILES.py]: I think the input would be given a pH, return the protonation state. However, I am not entirely sure what is needed for simulation. Could I just enumerate every protonation state?

Comment (pgrinaway) [reply Ligandr/NameToSMILES.py]: Ah, ok, I see. I would imagine you would actually need a pKa prediction to do what you'd like. Maybe Epik is appropriate here? I think you'd need an actual model of pKas.

Comment (404random) [reply Ligandr/NameToSMILES.py]: Ok, I'll look into that! Right now, just going to focus on reaching minimum functionality and then adapting the code later.

Comment (rafwiewiora) [reply Ligandr/NameToSMILES.py]: Yeah I think for now we'll be fine deciding what the protonation state should be "manually" (i.e. this is obvious for co-factors at least) and inputting the SMILES for the protonated ligand.

**Pull request merged**